src/murmur/Messages.cpp

```diff
@@ -155,6 +155,61 @@ bool isChannelEnterRestricted(Channel *c) {
 	return false;
 }
 
+/// Process the Capabilities message, used to negotiate voice protocol.
+/// See the doc in Mumble.proto for details about this message.
+///
+/// @param uSource ServerUser that receives this message.
+/// @param msg The message itself.
+void Server::msgCapabilities(ServerUser *uSource, MumbleProto::Capabilities &msg) {
+	MSG_SETUP(ServerUser::Connected);
+	int custom_index = 0;
+	for (int i = 0; i < msg.supported_protocols_size(); ++i){
+		auto protocol_type = msg.supported_protocols(i);
```

Comment:
I think using a const-reference here might be better than copying the data...

Reply:
Because this message can also be used to re-negotiate the protocol.